src/page-modules/assistant/layout.tsx

```diff
@@ -17,76 +17,61 @@ import { AnimatePresence, motion } from 'framer-motion';
+import { FromToTripQuery } from './types';
 import { createTripQuery } from './utils';
-import { MessageBox } from '@atb/components/message-box';
 
 export type AssistantLayoutProps = PropsWithChildren<{
-  initialFromFeature?: GeocoderFeature;
-  initialToFeature?: GeocoderFeature;
-  initialTransportModesFilter?: TransportModeFilterOption[] | null;
-  initialSearchTime?: SearchTime;
+  tripQuery: FromToTripQuery;
 }>;
 
-function AssistantLayout({
-  children,
-  initialFromFeature,
-  initialToFeature,
-  initialTransportModesFilter,
-  initialSearchTime,
-}: AssistantLayoutProps) {
+function AssistantLayout({ children, tripQuery }: AssistantLayoutProps) {
   const { t } = useTranslation();
   const router = useRouter();
 
   const [showAlternatives, setShowAlternatives] = useState(false);
-  const [selectedFromFeature, setSelectedFromFeature] = useState<
-    GeocoderFeature | undefined
-  >(initialFromFeature);
-  const [selectedToFeature, setSelectedToFeature] = useState<
-    GeocoderFeature | undefined
-  >(initialToFeature);
   const [searchTime, setSearchTime] = useState<SearchTime>(
-    initialSearchTime ?? { mode: 'now' },
+    tripQuery.searchTime,
   );
   const [transportModeFilter, setTransportModeFilter] = useState(
-    getInitialTransportModeFilter(initialTransportModesFilter),
+    getInitialTransportModeFilter(tripQuery.transportModeFilter),
   );
   const [isSwapping, setIsSwapping] = useState(false);
   const [isSearching, setIsSearching] = useState(false);
   const [geolocationError, setGeolocationError] = useState<string | null>(null);
 
-  const onSwap = async () => {
-    if (!selectedToFeature || !selectedFromFeature) return;
-    setIsSwapping(true);
+  const setValuesWithLoading = async (override: Partial<FromToTripQuery>) => {
+    setIsSearching(true);
     const query = createTripQuery(
-      selectedToFeature,
-      selectedFromFeature,
-      searchTime,
+      {
+        ...tripQuery,
+        ...override,
+        searchTime,
+      },
       transportModeFilter,
     );
-    const temp = selectedFromFeature;
-    setSelectedFromFeature(selectedToFeature);
-    setSelectedToFeature(temp);
-    setIsSwapping(false);
-
-    setIsSearching(true);
-    await router.push({ pathname: '/assistant', query });
+    await router.push({ query });
     setIsSearching(false);
   };
 
+  const onSwap = async () => {
+    setIsSwapping(true);
+    await setValuesWithLoading({
+      from: tripQuery.to,
+      to: tripQuery.from,
+    });
+    setIsSwapping(false);
+  };
+
   const onSubmitHandler: FormEventHandler<HTMLFormElement> = async (e) => {
     e.preventDefault();
-    if (!selectedFromFeature || !selectedToFeature) return;
-    setIsSearching(true);
-    const query = createTripQuery(
-      selectedFromFeature,
-      selectedToFeature,
-      searchTime,
-      transportModeFilter,
-    );
-    await router.push({ pathname: '/assistant', query });
-    setIsSearching(false);
+    setValuesWithLoading({});
```

Comment:
Why empty object here?

Reply:
Could be undefined, but feel just as OK to have empty override i feel to avoid the undefined stuff.